lib/rules/lint-required-context-role.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const AstNodeInfo = require('../helpers/ast-node-info');
+const Rule = require('./base');
+
+const errorMessage = (role, validParents) => {
+  return `You have an element with the role of "${role}" but it is missing the required (immediate) parent element of "[${validParents.join(
+    ', '
+  )}]". Reference: https://www.w3.org/TR/wai-aria-1.0/roles#${role}.`;
+};
+
+const ROOTS = {
+  row: ['rowheader', 'columnheader', 'gridcell'],
+  group: ['listitem', 'menuitem', 'menuitemradio', 'treeitem'],
+  menu: ['menuitem', 'menuitemcheckbox', 'menuitemradio'],
+  menubar: ['menuitem', 'menuitemcheckbox', 'menuitemradio'],
+  list: ['listitem'],
+  listbox: ['option'],
+  grid: ['row', 'rowgroup'],
+  rowgroup: ['row'],
+  treegrid: ['row'],
+  tablist: ['tab'],
+  tree: ['treeitem'],
+};
+
+const CHILDS_FOR_ANALYZE = [].concat.apply([], Object.keys(ROOTS).map(keyName => ROOTS[keyName]));
+const VALID_PARENTS_FOR_CHILD = Object.keys(ROOTS).reduce((result, keyName) => {
+  const childs = ROOTS[keyName];
+  childs.forEach(childName => {
+    if (!(childName in result)) {
+      result[childName] = [];
+    }
+    result[childName].push(keyName);
+  });
+  return result;
+}, {});
+Object.keys(VALID_PARENTS_FOR_CHILD).forEach(key => {
+  VALID_PARENTS_FOR_CHILD[key].sort();
+});
```

Comment:
we might want to add a Jest snapshot test to ensure that `VALID_PARENTS_FOR_CHILD` matches our expectations 🤔